ipa-core/src/protocol/ipa_prf/mod.rs

```diff
@@ -40,6 +41,119 @@ pub(crate) enum Step {
     SortByTimestamp,
 }
 
+#[derive(Debug)]
+#[cfg_attr(test, derive(Clone, PartialEq, Eq))]
+pub struct OPRFIPAInputRow<BK: SharedValue, TV: SharedValue, TS: SharedValue> {
+    pub match_key: Replicated<BA64>,
+    pub is_trigger: Replicated<Boolean>,
+    pub breakdown_key: Replicated<BK>,
+    pub trigger_value: Replicated<TV>,
+    pub timestamp: Replicated<TS>,
+}
+
+impl Serializable for u64 {
+    type Size = U8;
+    type DeserializationError = Infallible;
+
+    fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+        let raw = &self.to_le_bytes()[..buf.len()];
+        buf.copy_from_slice(raw);
+    }
+
+    fn deserialize(buf: &GenericArray<u8, Self::Size>) -> Result<Self, Self::DeserializationError> {
+        let mut buf_to = [0u8; 8];
+        buf_to[..buf.len()].copy_from_slice(buf);
+        Ok(u64::from_le_bytes(buf_to))
+    }
+}
```

Comment:
deserialization for u64 is straightforward - you can use `from_le_bytes`  from std. I'd suggest to remove this